Resources/config/messenger.xml

```diff
@@ -44,7 +44,6 @@
         -->
         <service id="messenger.transport.doctrine.factory" class="Symfony\Component\Messenger\Bridge\Doctrine\Transport\DoctrineTransportFactory" public="false">
             <argument type="service" id="doctrine" />
-            <tag name="messenger.transport_factory" />
```

Comment:
Wouldn't it be better to define as much things as we can here and remove the whole definition if criteria don't match? As far as I can see, at the moment you are leaving this service registered, but is unused because it won't have a tag.

Reply:
Probably not much of a difference between both approaches, since, from my understanding, the unused service will be removed during compilation. So I just went with the previous approach. But sure, I could reverse it if you want?

Reply:
Sure, it doesn't make huge difference, but it will be less work for symfony optimizer if we remove it ourselves. Plus it's following existing approach we use. We try to define all the tags in xml files and if it cannot be used, we remove whole service definition. I think it's also more clean approach, because you expect definitions we define in xml files to be as complete as possible, otherwise it's required to crosscheck them with with what's defined in Extension, which will get messy fast.

Reply:
I started working on it before you replied and I agree, its cleaner now 👍